docs/_tutorials/orderofinterpretation.md

```diff
@@ -0,0 +1,133 @@
+
+In `_data/mydata.yml`, suppose you have this mapping:
+
+```
+{% raw %}somevalue: {{myvar}}{% endraw %}
+```
+
+On a page, you try to insert this value:
+
+```
+{% raw %}{{site.data.mydata.somevalue}}{% endraw %}
+```
+Nothing renders because first the site variables populate, and then Liquid renders. When the site variables populate, the value for `{% raw %}{{site.data.mydata.somevalue}}{% endraw %}` is simply `{% raw %}{{myvar}}{% endraw %}`, which registers as a string and gets printed as a string. You can't use Liquid in data files.
+
+Similarly, suppose you have a `highlight` tag in your `_data/mydata.yml` file:
+
+```
+{% raw %}myvalue: >
+  {% highlight javascript %}
+  console.log('alert');
+  {% endhighlight %}{% endraw %}
+```
+
+On a page, you try to insert the value:
+
+```
+{% raw %}{{site.data.mydata.myvalue}}{% endraw %}
+```
+
+This renders as a string for the same reasons described above. When the `site.data.mydata.myvalue` tag populates, the value gets stored as a string and printed to the page as a string.
+
+To make the code work, consider populating content from includes.
+
+### Markdown in include file not processed
+
+Suppose you have a Markdown file at `_includes/mycontent.md`. In the Markdown file, you have some Markdown formatting:
+
+```markdown
+This is a list:
+* first item
+* second item
+```
+
+You include the file into an HTML file as follows:
+
+```liquid
+{% raw %}{% include mycontent.md %}{% endraw %}
+```
+
+The Markdown is not processed because first the Liquid (`include` tag) gets processed, inserting `mycontent.md` into the HTML file. *Then* the Markdown would get processed.
+
+But because the content is included into an *HTML* page, the Markdown isn't rendered. The Markdown filter processes content only in Markdown files.
+
+To make the code work, use HMTL formatting in includes inserted into HTML files.
+
+However, note that `highlight` tags don't require Markdown to process. Suppose your include contains the following:
+
+```liquid
+{% raw %}{% highlight javascript %}
+console.log('alert');
+{% endhighlight %}{% endraw %}
+```
+
+The `highlight` tag *is* Liquid. (Liquid passes the content to Rouge or Pygments for syntax highlighting.) As a result, this code will actually convert to HTML with syntax highlighting. Jekyll does not need the Markdown filter to process `highlight` tags.
+
+### Liquid mixed with JavaScript isn't rendered
+
+Suppose you try to mix Liquid with JavaScript, like this:
+
+```liquid
+{% raw %}if page.type == "home" { 
+    $("intro").addClass("bright");
+elsif page.type == "normal" {
+    $("intro").addClass("muted");
+ }{% endraw %}
+```
+
+The Liquid renders long before the page gets converted to HTML. So when JavaScript executes in the browser, the Liquid is no longer present.
+
+You can never mix Liquid with JavaScript because Liquid processes when the site builds. In contrast, JavaScript processes in the browser when a user interacts with your site.
```

Comment:
I understand what you're trying to say, but perhaps some different language could be used. Mixing Liquid and JavaScript might be Expert Mode, but it's not impossible.